src/components/Home/NewsletterPopup.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { Button } from "@/components/Button";
+
+interface NewsletterPopupProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+function setLocalStorage(key: string, days: number) {
+  const now = new Date();
+  const item = { expireDate: now.getTime() + days * 864e5 };
```

Comment:
Since we're just storing the expire date, no need to store it as a JSON object. Just store the time